commit be7be5da2a823ac2ce8826c695c6a56afdc17b33
Author: nao20010128nao <nao20010128@gmail.com>
Date:   Sat Jul 31 06:44:58 2021 +0900

    plugin.api.http_session: do not fail even if timeout is not set

diff --git a/src/streamlink/plugin/api/http_session.py b/src/streamlink/plugin/api/http_session.py
index 6cd3764..18bdaa8 100644
--- a/src/streamlink/plugin/api/http_session.py
+++ b/src/streamlink/plugin/api/http_session.py
@@ -134,7 +134,7 @@ class HTTPSession(Session):
         raise_for_status = kwargs.pop("raise_for_status", True)
         schema = kwargs.pop("schema", None)
         session = kwargs.pop("session", None)
-        timeout = kwargs.pop("timeout", self.timeout)
+        timeout = kwargs.pop("timeout", getattr(self, 'timeout', None))
         total_retries = kwargs.pop("retries", 0)
         retry_backoff = kwargs.pop("retry_backoff", 0.3)
         retry_max_backoff = kwargs.pop("retry_max_backoff", 10.0)
